services/dy-3dvis/Makefile

```diff
@@ -1,139 +1,205 @@
+	@$(call _docker_compose_build)
+
+define show-meta
+	$(foreach iid,$(shell docker images */$(1):* -q | sort | uniq),\
+		docker image inspect $(iid) | jq '.[0] | .RepoTags, .ContainerConfig.Labels, .Config.Labels';)
+endef
+
+info-build: ## displays info on the built image
+	# Built images
+	@docker images */$(DOCKER_IMAGE_NAME):*
+	@docker images */$(DOCKER_IMAGE_NAME)-gpu:*
+	# Tags and labels
+	@$(call show-meta,$(DOCKER_IMAGE_NAME))
+	@$(call show-meta,$(DOCKER_IMAGE_NAME)-gpu)
+
+# TESTS----------------------------------- 
+.PHONY: tests tests-unit tests-integration
+tests-unit tests-integration: ## runs integration and unit tests
+	@.venv/bin/pytest -vv \
+		--basetemp=$(CURDIR)/tmp \
+		--exitfirst \
+		--failed-first \
+		--pdb \
+		--junitxml=pytest_$(subst tests-,,$@)test.xml \
+		$(CURDIR)/tests/$(subst tests-,,$@)
+
+tests: tests-unit tests-integration ## runs unit and integration tests
+
+
+# PUBLISHING ----------------------------------- 
+define _bumpversion
+	# upgrades as $(subst $(1),,$@) version, commits and tags
+	@bump2version --verbose --list --config-file $(1) $(subst $(2),,$@)
+endef
+
+.PHONY: version-service-patch version-service-minor version-service-major
+version-service-patch version-service-minor version-service-major: versioning/service.cfg ## kernel/service versioning as patch
+	@$(call _bumpversion,$<,version-service-)
+
+.PHONY: push push-force push-version push-latest pull-latest pull-version tag-latest tag-version
+tag-latest tag-version:
+	docker tag local/$(DOCKER_IMAGE_NAME):production $(DOCKER_REGISTRY)/$(DOCKER_IMAGE_NAME):$(if $(findstring version,$@),$(DOCKER_IMAGE_TAG),latest)
+	docker tag local/$(DOCKER_IMAGE_NAME)-gpu:production $(DOCKER_REGISTRY)/$(DOCKER_IMAGE_NAME)-gpu:$(if $(findstring version,$@),$(DOCKER_IMAGE_TAG),latest)
+
+version_valid = $(shell test $$(echo $(DOCKER_IMAGE_TAG) | cut --fields=1 --delimiter=.) -gt 0 > /dev/null && echo "image version is valid")
+version_exists = $(shell DOCKER_CLI_EXPERIMENTAL=enabled docker manifest inspect $(DOCKER_REGISTRY)/$(1) > /dev/null && echo "image already exists on $(DOCKER_REGISTRY)")
+push push-force: ## pushes (resp. force) services to the registry if service not available in registry.
+	@$(if $(findstring force,$@),,\
+		$(if $(call version_valid),$(info version is valid), $(error $(DOCKER_IMAGE_TAG) is not a valid version (major>=1)))\
+		$(if $(call version_exists,$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG)),$(error $(DOCKER_REGISTRY)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG) already exists on $(DOCKER_REGISTRY)), $(info no version found on $(DOCKER_REGISTRY)))\
+		$(if $(call version_exists,$(DOCKER_IMAGE_NAME)-gpu:$(DOCKER_IMAGE_TAG)),$(error $(DOCKER_REGISTRY)/$(DOCKER_IMAGE_NAME):$(DOCKER_IMAGE_TAG) already exists on $(DOCKER_REGISTRY)), $(info no version found on $(DOCKER_REGISTRY)))\
+	)
+	@$(MAKE) push-version;

```

Comment:
MINOR: in order to reduce unnecessary noice, I have some verbose from these makes by using
```bash
$(MAKE) --no-print-directory 
```

Reply:
yep.. good point, I forgot these